### PR TITLE
Feature/branch details

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Enrich branch overview with more details like last committer and ahead/behind commits. Since calculating this information is pretty intense, we request it in chunks to prevent very long loading times. Also we cache the results in frontend and backend. 

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] New code is covered with unit tests
- [ ] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Feature has been tested with different permissions
- [x] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitBranchDetailsCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  public void shouldCountAheadAndBehind() {
+    GitBranchDetailsCommand command = new GitBranchDetailsCommand(createContext());
+
+    BranchDetailsCommandRequest request = new BranchDetailsCommandRequest();
+    request.setBranchName("test-branch");
+    BranchDetailsCommandResult result = command.execute(request);
+
+    assertThat(result.getChangesetsAhead()).isEqualTo(1);
+    assertThat(result.getChangesetsBehind()).isEqualTo(2);
+  }
```
Comment (eheimbuch) [inline scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitBranchDetailsCommandTest.java:44]: More tests?

File: scm-plugins/scm-git-plugin/src/main/java/sonia/scm/repository/spi/GitBranchDetailsCommand.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+public class GitBranchDetailsCommand extends AbstractGitCommand implements BranchDetailsCommand {
+
+  @Inject
+  GitBranchDetailsCommand(GitContext context) {
+    super(context);
+  }
+
+  @Override
+  public BranchDetailsCommandResult execute(BranchDetailsCommandRequest branchDetailsCommandRequest) {
+    String defaultBranch = context.getConfig().getDefaultBranch();
```
Comment (eheimbuch) [inline scm-plugins/scm-git-plugin/src/main/java/sonia/scm/repository/spi/GitBranchDetailsCommand.java:49]: Return `0,0` if requested branch equals default branch?

File: scm-ui/ui-webapp/src/repos/branches/components/AheadBehindTag.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      lengthInPercent += part / 5;
+    }
+    if (part < 500) {
+      lengthInPercent += part / 10;
+    }
+    if (part > 500) {
+      lengthInPercent += part / 25;
+    }
+    return lengthInPercent;
+  };
```
Comment (eheimbuch) [inline scm-ui/ui-webapp/src/repos/branches/components/AheadBehindTag.tsx:83]: What do you think about this calculation?

File: scm-webapp/src/main/resources/config/gcache.xml [hunk 4]
```diff
@@ -86,6 +86,15 @@
...
   />
 
+  <!--
+  Search cache for branch details
+-->
+  <cache
+    name="sonia.cache.cmd.branch-details"
+    maximumSize="10000"
+    expireAfterAccess="60000"
+  />
```
Comment (eheimbuch) [inline scm-webapp/src/main/resources/config/gcache.xml:96]: What do you think about this cache? It seems too big for me, but I have no idea what an appropriate size might be.

File: scm-ui/ui-webapp/src/repos/branches/components/AheadBehindTag.tsx (hunk 3 above)
Comment (pfeuffer) [reply scm-ui/ui-webapp/src/repos/branches/components/AheadBehindTag.tsx:83]: Indeed I thought about this a bit and played around with sqrt and log and so on, but could not find something where I had a 💡 moment.

What I would change is that I'd subtract the amount from `part` that already was considered like this, if this makes sense:

```java
    if (part < 10) {
      lengthInPercent += part;
    }
    if (part < 50) {
      lengthInPercent += (part - 10) / 5;
    }
    if (part < 500) {
      lengthInPercent += (part - 60) / 10;
    }
    if (part > 500) {
      lengthInPercent += (part - 560) / 25;
    }
```

File: scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitBranchDetailsCommandTest.java (hunk 1 above)
Comment (pfeuffer) [reply scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitBranchDetailsCommandTest.java:57]: Jep, got 'em

**Pull request merged**